packages/react/src/tests/e2e.test.tsx

```diff
@@ -391,4 +415,94 @@ describe('@remote-ui/react', () => {
     );
     expect(appElement.innerHTML).toBe('<img src="/image2.jpg">');
   });
+
+  describe('Remote tree updates', () => {
+    it('updates the remote tree when setting state with an array', () => {
+      const receiver = createRemoteReceiver();
+      const remoteRoot = createRemoteRoot(receiver.receive, {
+        components: [RemoteBox],
+      });
+
+      function RemoteApp() {
+        const [items, setItems] = useState(['a', 'b', 'c']);
+        const handleClick = useCallback(() => {
+          setItems((prev) => prev.reverse());
```

Comment:
FYI: This does not create a new array. The [Array.prototype.reverse()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reverse) method mutates the array in place. This can be verified in the console:

```js
const a = [1, 2, 3];
console.log(a === a.reverse()); // Logs: true
```

This is the reason why no state update is triggered. The setter functions use a referential equality check under the hood (`===`) and due to `.reverse()` mutating the array in place the setter receives the same array as it already had and cancels the state update.

Reply:
Thanks!  I deleted a test which was no longer relevant after this fix.  I still think there could be an issue with rendering duplicate nodes on the host side of the tree, though.  When I make this change, it renders:

```
<div>a</div>
<div>b</div>
<div>c</div>
<div>b</div>
<div>a</div>
```